src/core/ddsc/src/dds_entity.c

```diff
@@ -403,14 +403,16 @@ dds_return_t dds_delete_impl_pinned (dds_entity *e, enum delete_impl_state delst
     ddsrt_mutex_unlock (&p->m_mutex);
   }
 
-  /* Do some specific deletion when needed.  Bootstrapping and its inverse are always a
-     tricky business, and here it is no different: deleting the pseudo-top-level object
-     tears down all kinds of stuff that is supposed to remain in existence (like the
-     entire platform abstraction) and so it must be the final call.  Thus, we rely on it
-     to call "dds_entity_final_deinit_before_free" and return a special error code. */
+  /* Do some specific deletion when needed */
   ret = dds_entity_deriver_delete (e);
   if (ret == DDS_RETCODE_NO_DATA)
-    ret = DDS_RETCODE_OK;
```

Comment:
I trust you know what you're doing here as I'm still unfamiliar with the *pseudo-top-level* mechanism.

Reply:
Trust but verify, always a good procedure. I have noticed there is still a problem with that "pseudo-top-level" mechanism in that a failed domain creation can result in an initialised library and that there are some cases where that would break things — they're edge cases, but that's not a reason to ignore them.

On reflection, I also disagree with the way delete and the entity reference count play together. If I call delete, the handle should be gone, simple as that. The only type of object that ever needs to linger is the topic (if it is still used in readers or writers), but even so, the handle can be closed. That confusion has been present from the initial commit, and it does play a role in the this edge case in dealing with this top-level object.

But I digress: the change here is because the analysers rightly complained that this is dead code, and so replacing it by a comment won't affect the behaviour. But there are other ways of addressing that issue, and the one where the assignment stays and ``ret`` is then actually used is better.

I'll change it.